frontends/p4/def_use.h

```diff
@@ -487,16 +560,17 @@ class ComputeWriteSet : public Inspector, public IHasDbPrint {
     const StorageMap *storageMap;
     /// if true we are processing an expression on the lhs of an assignment
     bool lhs;
-    /// For each expression the location set it writes
-    hvec_map<const IR::Expression *, const LocationSet *> writes;
+    /// For each program location the location set it writes
+    ordered_map<loc_t, const LocationSet *> writes;
```

Comment:
@asl Let me know if there is a way to use `hvec_map` here instead of `ordreed_map`. I reverted to `ordered_map` because I ran into a bunch of errors when changing the key to `loc_t`.

Reply:
`ordered_map` is terrible expensive. And this is one of the most hottest place. I would probably suggest that every PR that contains changes to use-def in this way to contain runtimes of:
```
test/gtestp4c --gtest_filter=P4CParserUnroll.switch_20160512
```

and
```
ctest -R "p14_to_16/testdata/p4_14_samples/switch_20160512/switch.p4"
```

They are more or less the same source code, just running a bit different set of passes.

Reply:
Results of running the above commands on this branch:

test/gtestp4c --gtest_filter=P4CParserUnroll.switch_20160512
```
7930 ms
7918 ms
7949 ms
```

ctest -R "p14_to_16/testdata/p4_14_samples/switch_20160512/switch.p4"
```
23.14 s
23.19 s
23.00 s
```

Main branch:

test/gtestp4c --gtest_filter=P4CParserUnroll.switch_20160512
```
7858 ms
7854 ms
7873 ms
```

ctest -R "p14_to_16/testdata/p4_14_samples/switch_20160512/switch.p4"
```
22.78 s
22.94 s
23.19 s
```

Reply:
> ordered_map is terrible expensive. And this is one of the most hottest places. I would probably suggest that every PR that contains changes to use-def in this way to contain runtimes of:

@asl changed to `hvec_map`